internal/statestore/backfill.go

```diff
@@ -143,6 +151,48 @@ func (rb *redisBackend) UpdateBackfill(ctx context.Context, backfill *pb.Backfil
 	return nil
 }
 
+// CleanupBackfills removes expired backfills
+func (rb *redisBackend) CleanupBackfills(ctx context.Context) error {
+	redisConn, err := rb.redisPool.GetContext(ctx)
+	if err != nil {
+		return status.Errorf(codes.Unavailable, "CleanupBackfills, failed to connect to redis: %v", err)
+	}
+	defer handleConnectionClose(&redisConn)
+
+	expiredBfIDs, err := rb.GetExpiredBackfillIDs(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, id := range expiredBfIDs {
+		_, tickets, err := rb.GetBackfill(ctx, id)
+		if err != nil {
+			logger.WithFields(logrus.Fields{
+				"error":       err.Error(),
+				"backfill_id": id,
+			}).Error("cleanup failed to get the backfill")
+			continue
+		}
+
+		err = rb.DeleteBackfill(ctx, id)
+		if err != nil {
+			logger.WithFields(logrus.Fields{
+				"error":       err.Error(),
+				"backfill_id": id,
+			}).Error("cleanup failed to delete the backfill")
```

Comment:
if you fail to delete the actual backfill (so it still has Tickets), do you still want to go ahead and delete tickets from Pending Release? would that not leave the system in inconsistent state?

Reply:
we try to delete as much as we can